code/game/objects/items/weapons/grenades/flashbang.dm

```diff
@@ -5,39 +5,42 @@
 	origin_tech = "materials=2;combat=3"
 	light_power = 10
 	light_color = LIGHT_COLOR_WHITE
-	var/light_time = 2
-	var/range = 7
+
+	var/light_time = 0.2 SECONDS // The duration the area is illuminated
+	var/range = 7 // The range in tiles of the flashbang
 
 /obj/item/grenade/flashbang/prime()
 	update_mob()
-	var/flashbang_turf = get_turf(src)
-	if(!flashbang_turf)
+	var/turf/T = get_turf(src)
+	if(!T)
 		return
 
-	set_light(7)
-
+	// VFX and SFX
 	do_sparks(rand(5, 9), FALSE, src)
-	playsound(flashbang_turf, 'sound/effects/bang.ogg', 25, 1)
-	bang(flashbang_turf, src, range)
+	playsound(T, 'sound/effects/bang.ogg', 100, TRUE)
+	new /obj/effect/dummy/lighting_obj(T, light_color, range + 2, light_power, light_time)
 
-	for(var/obj/structure/blob/B in hear(8, flashbang_turf))     		//Blob damage here
-		var/damage = round(30 / (get_dist(B, get_turf(src)) + 1))
-		B.take_damage(damage, BURN, "melee", 0)
+	// Stunning mechanic
+	bang(T, src, range)
 
-	spawn(light_time)
-		qdel(src)
+	// Blob damage here
+	for(var/obj/structure/blob/B in hear(range + 1, T))
+		var/damage = round(30 / (get_dist(B, T) + 1))
+		B.take_damage(damage, BURN, "melee", FALSE)
+
+	qdel(src)
```

Comment:
On an unrelated sidenote. Can we add this to sonic powder or flash powder or whichever reagent affects the blob IC? I would love to see all the wonderful creations people make to crowd control the blob

Reply:
Donezo